CX20x0, CX50xx, and CX51xx support with console/GUI image generation tool

klauer opened the pull request:
Supports CX51xx, CX50xx, CX20x0 (untested).
Adds a GUI for creating images:

macOS:
<img width="420" alt="image" src="https://user-images.githubusercontent.com/5139267/135899217-91395fa5-b64f-4493-af28-9d08b7487325.png">

Windows:

<img width="399" alt="image" src="https://user-images.githubusercontent.com/5139267/135906082-b45f7dfc-25ee-4128-ac4e-15aba2690911.png">

With some minimal input checking.

klauer commented:
Confirmed working by Vincent, with a few tweaks to the installation instructions.
I think this could/should be expanded to other PLC models now.

ZLLentz reviewed (approved):
My lack of familiarity with tk, with the plc regfiles, and with the overall goals of the project makes it hard to give a review here, but I didn't see any red flags. Docstrings are a bit light for my taste.

klauer commented:
Thanks, Zach! I cleaned things up just a bit and added some docstrings ("light on docstrings" definitely got a chuckle from me 😆 ) and will call this good enough for now. Will see how it fares later.
